bastion/gui/dashboard/main_window.py

```
@@ -0,0 +1,813 @@
+
+            # Store key in user data for deletion using setData(Qt.UserRole, ...) on first item
+            self.table_rules.item(row, 0).setData(Qt.ItemDataRole.UserRole, key)
+
+    def refresh_logs(self):
+        def load_logs():
+            try:
+                # Check if we can read the log file directly
+                if os.access(self.log_path, os.R_OK):
+                    # We can read it directly
+                    cmd = ['tail', '-n', '50', str(self.log_path)]
+                    res = subprocess.run(cmd, capture_output=True, text=True, timeout=10)
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)